src/main/java/de/tum/in/www1/artemis/web/rest/GradingInstructionResource.java

```diff
@@ -0,0 +1,151 @@
+        }
+
+        gradingInstruction = gradingInstructionService.save(gradingInstruction);
+        return ResponseEntity.created(new URI("/api/grading-instruction/" + gradingInstruction.getId()))
+                .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, gradingInstruction.getId().toString())).body(gradingInstruction);
+
+    }
+
+    /**
+     * PUT /grading-instruction : Updates an existing gradingInstruction.
+     *
+     * @param gradingInstruction the gradingInstruction to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated gradingInstruction, or with status 400 (Bad Request) if the gradingInstruction is not valid, or with status 500
+     * (Internal Server Error) if the gradingInstruction couldn't be updated
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PutMapping("/grading-instruction")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<GradingInstruction> updateGradingInstruction(@RequestBody GradingInstruction gradingInstruction) throws URISyntaxException {
+        log.debug("REST request to update GradingInstruction : {}", gradingInstruction);
+        if (gradingInstruction.getId() == null) {
+            return createGradingInstruction(gradingInstruction);
+        }
+
+        // fetch exercise from database to make sure client didn't change groups
+        Exercise exercise = exerciseService.findOne(gradingInstruction.getExercise().getId());
+        if (exercise == null) {
+            return ResponseEntity.badRequest().headers(
+                    HeaderUtil.createFailureAlert(applicationName, true, ENTITY_NAME, "exerciseNotFound", "The exercise belonging to this grading instruction does not exist"))
+                    .body(null);
+        }
+        gradingInstruction = gradingInstructionService.save(gradingInstruction);
+
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, gradingInstruction.getId().toString())).body(gradingInstruction);
+    }
+
+    /**
+     * DELETE /grading-instruction/:gradingInstructionId : delete the "id" gradingInstruction.
+     *
+     * @param gradingInstructionId the id of the gradingInstruction to delete
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @DeleteMapping("/grading-instruction/{gradingInstructionId}")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Void> deleteGradingInstruction(@PathVariable Long gradingInstructionId) {
+        log.debug("REST request to delete GradingInstruction : {}", gradingInstructionId);
+
+        GradingInstruction gradingInstruction = gradingInstructionService.findOne(gradingInstructionId);
+        Exercise exercise = exerciseService.findOne(gradingInstruction.getExercise().getId());
+        if (Optional.ofNullable(gradingInstruction).isPresent()) {
```

Comment:
What if this condition is false? At the moment the endpoint would still return ok as if deletion was successful. And wouldn't there be an issue in the previous line if `gradingInstruction` were null?

Reply:
This is not resolved. You didn't change anything here.